client/app/collections/contacts.coffee

```diff
@@ -1,32 +1,45 @@
 ContactsListener = require 'lib/contacts_listener'
+Contact = require 'models/contact'
 
 
-comparator = (sort) ->
-    fn = (a, b) ->
-        [a, b] = [a, b].map (model) ->
-            name = model.get('n').split(';')
-            if fn.sort is 'gn'
-                "#{name[0]}#{name[1]}"
+
+module.exports = class Contacts extends Backbone.Collection
+
+    model: Contact
+    url: 'contacts'
+
+    sortDisabled: false
+
+    buildComparator: (sort) =>
+        fn = (a, b) =>
+            if @sortDisabled
+                return -1
             else
-                "#{name[1]}#{name[0]}"
+                [a, b] = [a, b].map (model) ->
```

Comment:
I first wrote this part. What we need is to return a string with _firstname, lastname_ or _lastname, firstname_, regarding the sort option value, and then compare it to determines the sort order. How can we do it more efficiently (extracting the string for comparison) in the two models?